### ISSUE TITLE
Executable crashes when packed with --ultra-brute

### ISSUE THREAD

**Issue by dacto** [upx/upx#385]
## What's the problem (or question)?

Executable crashes when packed using `--ultra-brute` option.

Note: executable is statically linked binary with musl.

```
->strace ultra-brute-demo
execve("/tmp/ultra-brute-demo", ["ultra-brute-demo"], [/* 108 vars */]) = 0
open("/proc/self/exe", O_RDONLY)        = 3
mmap(NULL, 335074, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f541651e000
mmap(0x7f541651e000, 334654, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED, 3, 0) = 0x7f541651e000
mprotect(0x7f541656e000, 7394, PROT_READ|PROT_EXEC) = 0
readlink("/proc/self/exe", "/tmp/ultra-brute-demo", 4095) = 21
exit(127)                               = ?
+++ exited with 127 +++
```

## What should have happened?

Executable packed with `--ultra-brute` option does not crash and stdout is `hello world!`.

## Do you have an idea for a solution?

Observation: executable does not crash when packed using `--best` option instead.

## How can we reproduce the issue?

I've attached a zip archive ([demo.zip](https://github.com/upx/upx/files/4914007/demo.zip)) containing:
|description|filename|
|------------|---------|
| original unpacked executable|`original-demo`|
| `--best` packed executable|`best-demo`|
| `--ultra-brute` packed executable|`ultra-brute-demo`|

1. pack original-demo executable using `--ultra-brute`: `upx --ultra-brute original-demo`
2. run packed executable and witness crash: `./original-demo`

## Please tell us details about your environment.

* UPX version used (`upx --version`): `upx 3.96` (also occurs with `upx 3.95`)
* Host Operating System and version: `Alpine Linux 3.9` (for musl static compile/link) within docker container on Target listed below
* Host CPU architecture: `x86_64`
* Target Operating System and version: `Ubuntu 16.04.6 LTS, Linux Kernel 4.4.0-173-generic`
* Target CPU architecture: `x86_64`

**Comment by markus-oberhumer** [comment]
Is this fixed in current git?

**Comment by dacto** [comment]
@markus-oberhumer
The demo executable now emits the expected output (`hello world!`) when packed with `--ultra-brute` from 8ee39aaa34f41d4db0c28be966a7cb35e392eb79 (with UCL).

For reference I've attached the demo packed using `--best` and `--ultra-brute` from 8ee39aaa34f41d4db0c28be966a7cb35e392eb79.
[demo-fixed-8ee39aa.zip](https://github.com/upx/upx/files/9438441/demo-fixed-8ee39aa.zip)

Thank you! :+1:

**Comment by dacto** [comment]
@markus-oberhumer are you planning a new release in the near future?
